library/data/loot_table/src/main/java/org/quiltmc/qsl/loot/api/QuiltLootPool.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016, 2017, 2018, 2019 FabricMC
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.loot.api;
+
+import java.util.List;
+
+import net.minecraft.loot.LootPool;
+import net.minecraft.loot.condition.LootCondition;
+import net.minecraft.loot.entry.LootPoolEntry;
+import net.minecraft.loot.function.LootFunction;
+import net.minecraft.loot.provider.number.LootNumberProvider;
+
+/**
+ * Interface implemented by {@link net.minecraft.loot.LootPool} instances when QSL is present.
+ *
+ * <p>Contains accessors for various fields.
+ *
+ * @see	QuiltLootPoolBuilder
+ * @see #cast(LootPool)
+ */
+public interface QuiltLootPool {
```

Comment:
Static accessor classes (with methods like `QuiltLootPools.getEntries(LootPool)`) might be nicer for modders to use, and don't require any sort of casting or implementing interfaces with mixin. (I opted for that approach in my largely forgotten loot table API v2 for Fabric API)